main.tf

```diff
@@ -1,7 +1,12 @@
+locals {
+  # Remove the var.domain_name from the subject_alternative_names as it doesn't need to be there.
+  subject_alternative_names = ["${concat( slice(var.subject_alternative_names, 0, index(var.subject_alternative_names, var.domain_name)), slice(var.subject_alternative_names, index(var.subject_alternative_names, var.domain_name) + 1 , length(var.subject_alternative_names) ))}"]
```

Comment:
Though it would be in the list as a result of user error, no? Or was there a logical error somewhere?

Reply:
If I understand this correct, in plain american english, this will remove the first occurrence of `var.domain_name` from a list of `var.subject_alternative_names`. 

If that's the case, seems to me we should also do `distinct(sort(...))`